dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+             */
+            var version = $.fn.jquery.split(' ')[0].split('.');
+            // main check, changed to better reading
+            if ((version[0] <= REQ_JQUERY_MAJOR) && (version[1] <= REQ_JQUERY_MINOR)) {
+                throw new Error('multiselect requires jQuery version ' + REQ_JQUERY_MAJOR + '.' + REQ_JQUERY_MINOR + ' or higher');
+            }
+        }
+
+        if (typeof define === 'function' && define.amd) {
+            // AMD. Register as an anonymous module depending on jQuery.
+            define(['jquery'], factory);
+            require(["jquery"], checkjQueryVersion);
+        } else {
+            // No AMD. Register plugin with global jQuery object.
+            checkForGlobaljQuery();
+            checkjQueryVersion(jQuery);
+            factory(jQuery);
+        }
+    }(function ($) {
+        'use strict';
+
+        /**
+         * Multiselect API.
+         * @module
+         */
+        var Multiselect = (function($) {
+            // private area - constants
+            /** Keycode for Enter.
+             * @type {number} */
+            const KEY_ENTER = 13;
+
+            /** Keycode for Backspace.
+             * @type {number} */
+            const KEY_BACKSPACE = 8;
+
+            /** Keycode for Del.
+             * @type {number} */
+            const KEY_DEL = 46;
+
+            /** User Agent identification for Internet Explorer up until IE 10.
+             * @type {string} */
+            const USER_AGENT_IE_UPTO_10 = "MSIE ";
+
+            /** User Agent identification for the Edge Browser.
+             * @type {string} */
+            const USER_AGENT_EDGE = "Edge/";
+
+            /** User Agent identification for Internet Explorer 11.
+             * @type {string} */
+            const USER_AGENT_IE_11 = "Trident/";
+
+            /** User Agent identification for the Safari browser.
+             * @type {string} */
+            const USER_AGENT_SAFARI = "safari";
+
+            /** CSS class applied to options that to filtered options.
+             * @type {string} */
+            const CSS_HIDDEN = "hidden";
+
+            /** Selector for the filtered option class.
+             * @type {string} */
+            const SELECTOR_HIDDEN = "." + CSS_HIDDEN;
+
+            /**
+             * Enum for the comparator results for sort functions.
+             * @readonly
+             * @enum {number} COMPARISON_RESULTS
+             */
+            const COMPARISON_RESULTS = {
+                /** The first parameter should be on a lower index than the second parameter */
+                A_COMES_FIRST: -1,
+                /** The second parameter should be on a lower index than the first parameter */
+                B_COMES_FIRST: 1,
+                /** The parameters are equal or just should not be moved around */
+                DO_NOT_CHANGE: 0
+            };
+
+            /** Data attribute used to store the position any select items had
```

Comment:
This is the old data attribute to store position. Extracted to constant because then it can be easily changed.